packages/manager/src/hooks/useGravatar.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from 'react';
+
+import { checkForGravatar, getGravatarUrl } from 'src/utilities/gravatar';
+
+// set a cache to prevent duplicate requests
+const gravatarCache: Record<string, boolean> = {};
```

Comment:
Could we use a `Map`? It might allow us to make it even cleaner!

```suggestion
const gravatarCache = new Map<string, boolean>();
```

Reply:
I still like Banks suggestion here, map and get/set method is nicer